ext/geos_c_impl/factory.c

```diff
@@ -339,25 +322,18 @@ static VALUE method_factory_write_for_psych(VALUE self, VALUE obj)
   VALUE result;
   char* str;
   char has_3d;
-#ifndef RGEO_GEOS_SUPPORTS_SETOUTPUTDIMENSION
-  RGeo_Globals* globals;
-  VALUE wkt_generator;
-#endif
 
   self_data = RGEO_FACTORY_DATA_PTR(self);
   self_context = self_data->geos_context;
   has_3d = self_data->flags & RGEO_FACTORYFLAGS_SUPPORTS_Z_OR_M;
 #ifndef RGEO_GEOS_SUPPORTS_SETOUTPUTDIMENSION
   if (has_3d) {
-    globals = self_data->globals;
-    wkt_generator = globals->psych_wkt_generator;
-    if (NIL_P(wkt_generator)) {
-      wkt_generator = rb_funcall(
-        rb_const_get_at(globals->geos_module, rb_intern("Utils")),
+    if (NIL_P(psych_wkt_generator)) {
+      psych_wkt_generator = rb_funcall(
```

Comment:
shouldn't this be marked as well ? Note that however, in the current case, I did not find a way to generate a segv.

I tried some variations of :

```ruby
RGeo::Geos::Utils.singleton_class.undef_method :psych_wkt_generator
RGeo::WKRep::WKTGenerator.undef_method :generate
RGeo::Geos::Utils = nil
f = RGeo::Geos.factory(has_z_coordinate: true)
pt = f.point(1, 0, 0);nil
GC.start
GC.verify_compaction_references(double_heap: true, toward: :empty)
f.write_for_psych(pt)

ObjectSpace.each_object(RGeo::WKRep::WKTGenerator).count # run this between every line to see evolution
```

At some point, I just found that `RGeo::Geos.capi_supported?` is false. It looks like there is an issue with our new organisation:

![image](https://user-images.githubusercontent.com/11378424/148533128-75958ce2-e1aa-4e3e-9674-8e1f9cfc9831.png)

Reply:
Effectively there was a need to mark this as well. However i think that since the value mutates, simply using `rb_gc_register_mark_object` was not sufficient (still segfaults using ur reproduction). By using `rb_gc_register_address` it seems to be better.